sdk/src/main/java/io/dapr/client/DaprClient.java

```diff
@@ -20,7 +20,7 @@
      * @param <T>   The type of event to be publishded.
      * @return a Mono plan of type Void
      */
-    <T> Mono<Void> publishEvent(T event);
+    <T> Mono<Void> publishEvent(ClientRequest<T> event);
```

Comment:
This seems to make the interface specific to http

Reply:
I did not found anything specific to GPRC that is needed beyond the actual value to be sent over, unlike for http who needs more data, do you have any suggestion on how to handle that?